src/components/GenerativeDirectAnswerDisplay.tsx

```diff
@@ -0,0 +1,137 @@
+import { 
+  GenerativeDirectAnswerResponse, 
+  useSearchActions, 
+  useSearchState, 
+  SearchTypeEnum, 
+  Result 
+} from '@yext/search-headless-react';
+import { useComposedCssClasses } from '../hooks';
+import { executeGenerativeDirectAnswer } from '../utils/search-operations';
+import ReactMarkdown from "react-markdown";
+import React from 'react';
+
+/**
+ * The CSS class interface used for {@link GenerativeDirectAnswerDisplay}.
+ *
+ * @public
+ */
+export interface GenerativeDirectAnswerDisplayCssClasses {
+  generativeDirectAnswerTitle?: string,
+  generativeDirectAnswerText: string,
+  generativeDirectAnswerDivider: string,
+  generativeDirectAnswerContainer?: string,
+  generativeDirectAnswerCitations: string,
+  generativeDirectAnswerCitation?: string,
+  generativeDirectAnswerCitationTitle?: string,
+  generativeDirectAnswerCitationSnippet?: string,
+}
+
+const builtInCssClasses: Readonly<GenerativeDirectAnswerDisplayCssClasses> = {
+  generativeDirectAnswerTitle: 'text-xl',
+  generativeDirectAnswerText: 'mt-4',
+  generativeDirectAnswerDivider: 'border-b border-gray-200 w-full pb-6 mb-6',
+  generativeDirectAnswerContainer: 'p-6 border border-gray-200 rounded-lg shadow-sm',
+  generativeDirectAnswerCitations: 'mt-4 flex overflow-x-auto gap-4',
+  generativeDirectAnswerCitation: 'p-4 border border-gray-200 rounded-lg shadow-sm bg-slate-100 flex flex-col basis-64 text-sm text-neutral',
+  generativeDirectAnswerCitationTitle: 'font-bold',
+  generativeDirectAnswerCitationSnippet: 'line-clamp-2 text-ellipsis break-words',
+};
+
+/**
+ * Props for {@link GenerativeDirectAnswerDisplay}.
+ *
+ * @public
+ */
+export interface GenerativeDirectAnswerDisplayProps<T> {
```

Comment:
what is the generic type for?

Reply:
I have renamed this to GenerativeDirectAnswerProps instead. It's the props to GDA component. Currently, its fields include the css classes, answer header, and citations header so user can customize the UI.

Reply:
I still don't follow what is the generic type (<T>) is for here.
<img width="626" alt="Screenshot 2024-06-13 at 6 43 13 PM" src="https://github.com/yext/search-ui-react/assets/36055303/82c6ec29-79e3-4023-ba17-63026841fbcc">
Unless I missed it somewhere, in your latest update, the type T is not use in any of the fields, so I believe you can remove this throughout GDA component